news/tests/test_views.py

```diff
@@ -565,6 +601,83 @@ def test_success_sync_optin_lowercase(self):
             self.update_user_task.assert_called_with(request, SUBSCRIBE, data=request_data,
                                                      optin=True, sync=True)
 
+    @patch('news.views.IP_RATE_LIMIT_EXTERNAL', '1/m')
+    def test_ip_rate_limit(self):
+        self.update_user_task.return_value = HttpResponseJSON({'status': 'ok'})
+        request_data = {'newsletters': 'news,lets', 'optin': 'N', 'sync': 'N',
+                        'email': 'dude@example.com'}
+        self.client.post('/news/subscribe/', request_data,
```

Comment:
I had it that way and the tests failed. I then realized it was because the middleware is what does the handling and sending to the `ratelimited` view. To test the whole thing it has to be like this.